clippy_lints/src/drop_bounds.rs

```diff
@@ -0,0 +1,61 @@
+use crate::utils::{match_def_path, paths, span_lint};
+use if_chain::if_chain;
+use rustc::hir::*;
+use rustc::hir::GenericBound::Trait;
+use rustc::lint::{LateLintPass, LintArray, LintPass};
+use rustc::{declare_tool_lint, lint_array};
+
+/// **What it does:** Checks for generics with `std::ops::Drop` as bounds.
+///
+/// **Why is this bad?** `Drop` bounds do not really accomplish anything.
+/// A type may have compiler-generated drop glue without implementing the
+/// `Drop` trait itself. The `Drop` trait also only has one method,
+/// `Drop::drop`, and that function is by fiat not callable in user code.
+/// So there is really no use case for using `Drop` in trait bounds.
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rust
+/// fn foo<T: Drop>() {}
+/// ```
+declare_clippy_lint! {
+    pub DROP_BOUNDS,
```

Comment:
cc @oli-obk is this named according to our guidelines?

(we should perhaps write these down somewhere)

Reply:
I believe the ones that we use are [these](https://rust-lang.github.io/rfcs/0344-conventions-galore.html#lints)?

Reply:
Yes, it fits the rules linked by @phansch